platforms/scripts/afu_synth_setup

```diff
@@ -237,6 +237,37 @@ def setup_build(args):
     cmd.append('--verilog-hdr=../hw/afu_json_info.vh')
     subprocess.check_call(cmd, cwd=build_dir)
 
+    # Handle Qsys IPX file discovery
+    setup_build_ipx(args, sources, build_dir)
+
+
+# Construct the Qsys components.ipx file in the build directory, if needed.
+def setup_build_ipx(args, sources, build_dir):
+    # Get IPX files from source list
+    cmd = ['rtl_src_config']
+    cmd.append('--ipx')
+    if (os.path.isabs(sources)):
+        cmd.append('--abs')
+        cmd.append(sources)
+    else:
+        cmd.append(os.path.relpath(sources, build_dir))
+    ipx = commands_list_getoutput(cmd, cwd=build_dir)
+    ipx = ipx.strip()
+    # Anything to do?
+    if (len(ipx) == 0):
+        return
+    ipx = ipx.split('\n')
+
+    # Emit components.ipx
+    with open(os.path.join(build_dir, 'components.ipx'), 'w') as f:
```

Comment:
I guess, this is fine for now. You' are likely run into issues if ACDS team changes the file format in future QII versions. Have you thought about that?

Reply:
How is that different from the current situation? We'd have to update either the AFU source (where there is a components.idx that gets copied into the build) or we can update this script. This script could respond dynamically to the Qsys version.

What would be best is if they stop requiring the file!  I'd think it could be derived, given that they are given a set of qsys and ip files in the project.

Reply:
I think your best bet would be to dynamically generate  .idx based on the Qsys version. You'll have to watch out for changes between releases. In my experience, changes to IP formats are not uncommon.